packages/paste-core/components/chat-composer/src/ChatComposer.tsx

```diff
@@ -93,18 +126,38 @@ export const ChatComposer = React.forwardRef<HTMLDivElement, ChatComposerProps>(
       placeholder = "",
       initialValue,
       config,
-      maxHeight,
+      maxHeight = "size30",
       disabled,
+      fontSize,
+      lineHeight,
+      editorInstanceRef,
       ...props
     },
     ref,
   ) => {
+    const { setIsDisabled } = React.useContext(ChatComposerContext);
+
     const baseConfigWithEditorState = {
       ...baseConfig,
       editable: disabled ? false : true,
       editorState: initialValue ? () => renderInitialText(initialValue) : undefined,
     };
 
+    const getDisabledStyling = React.useCallback((): BoxStyleProps => {
+      /**
+       * If setIsDisabled is defined, then the styling will be handled by ChatComposerContainer.
+       * If it is not defined, then the styling will be handled by ChatComposer. Using both causes the diabled style tochange
```

Comment:
question: what do you mean when you say that using both causes the disabled style to change from container and then composer? just want to make sure I understand

Reply:
Exactly that. With the double styling it get's delayed when the disabled prop changes and the user sees the container grey area removed then the Lexical rich text component style change